frontend/talentsearch/src/js/components/roleSalaryForm/RoleSalaryForm.tsx

```diff
@@ -0,0 +1,351 @@
+      </span>
+    );
+  }
+  function modalThreeAdvisor(msg: string) {
+    return (
+      <span
+        role="dialog"
+        onClick={(e) => {
+          setDialogLevel3AdvisorOpen(true);
+          e?.preventDefault();
+        }}
+      >
+        {msg}
+      </span>
+    );
+  }
+  function modalFourManager(msg: string) {
+    return (
+      <span
+        role="dialog"
+        onClick={(e) => {
+          setDialogLevel4ManagerOpen(true);
+          e?.preventDefault();
+        }}
+      >
+        {msg}
+      </span>
+    );
+  }
+  function modalFourAdvisor(msg: string) {
+    return (
+      <span
+        role="dialog"
+        onClick={(e) => {
+          setDialogLevel4AdvisorOpen(true);
+          e?.preventDefault();
+        }}
+      >
+        {msg}
+      </span>
+    );
+  }
+
+  return (
+    <ProfileFormWrapper
+      title={intl.formatMessage({
+        defaultMessage: "Role and Salary Expectations",
+        description: "Title role and salary expectations form",
+      })}
+      description={intl.formatMessage({
+        defaultMessage:
+          "Government classifications are labels that the Government of Canada uses to group similar types of work. In the Government of Canada salary is tied to how positions are classified.",
+        description: "Description for the role and salary expectation form",
+      })}
+      crumbs={[
+        {
+          title: intl.formatMessage({
+            defaultMessage: "Role and Salary Expectations",
+            description: "Label for role and salary link",
+          }),
+        },
+      ]}
+    >
+      <BasicForm
+        onSubmit={handleSubmit}
+        options={{
+          defaultValues: initialDataToFormValues(initialUser),
+        }}
+      >
+        <p data-h2-margin="b(bottom, l)">
+          {intl.formatMessage(
+            {
+              defaultMessage:
+                "This platform is focused on hiring digital talent to work in positions classified as IT(Information Technology). Look at the following levels within the IT classification and <bold>select only</bold> the ones that represent the work you want to do.",
+              description: "Blurb describing the purpose of the form",
+            },
+            { bold },
+          )}
+        </p>
+        <Checklist
+          idPrefix="roleSalary"
+          legend={intl.formatMessage({
+            defaultMessage:
+              "I would like to be referred for jobs at the following levels:",
+            description: "Legend for role and salary checklist form",
+          })}
+          rules={{
+            required: intl.formatMessage(errorMessages.required),
+          }}
+          name="roleSalary"
+          items={[
+            {
+              value: 1,
+              label: intl.formatMessage(
+                {
+                  defaultMessage:
+                    "Level 1: Technician ($60,000 to $78,000). <modalOne><underline>Learn about IT-01</underline></modalOne>",
```

Comment:
I don't love this.  It's janky to have to wrap two elements around this and in my browser I don't get the pointer cursor to say it's a link.  I'm guessing this is the reason for the a11y overrides, too?  Is there any way to improve this?

Reply:
It looks like click handlers were applied to `<span>` tags which is not permitted. You will need to convert them to `<button>` elements and style them to appear like an `<a>`.

Reply:
OK, this is much much better.  I have a few more ideas to streamline it though. :smile: 

1) If you move the underline styling right into your model button you can get ride of the extra `underline` function and elements in your intl strings.
![image](https://user-images.githubusercontent.com/8978655/169573432-7dacfbb9-d224-4bc7-9c0c-6e54e663fb8f.png)

2) If you update `modalOne` to accept a function parameter you can reuse the same function for all your strings and get rid of the rest.
![image](https://user-images.githubusercontent.com/8978655/169573644-a62f762e-3129-482e-b789-ee2e3d3958d4.png)
![image](https://user-images.githubusercontent.com/8978655/169573728-7997638b-d6df-4936-9a43-c8c1d2433cc3.png)

3) I know you're only using it once but I'd recommend generalizing the link formatting as well:
![image](https://user-images.githubusercontent.com/8978655/169574318-d22e5b3d-2bd4-4fe5-9c73-0e8d7bebb515.png)
![image](https://user-images.githubusercontent.com/8978655/169574393-ccedca5c-bdc0-40df-82da-e7319728fc1f.png)

Reply:
😲